app/assets/javascripts/admin/customers/directives/edit_address_dialog.js.coffee

```diff
@@ -0,0 +1,36 @@
+angular.module("admin.customers").directive 'editAddressDialog', ($compile, $templateCache, $filter, DialogDefaults, Customers, StatusMessage) ->
+  restrict: 'A'
+  scope: true
+  link: (scope, element, attr) ->
+    scope.errors = []
+
+    scope.$watch 'address.country_id', (newVal) ->
+      if newVal
+        scope.states = scope.filter_states(newVal)
+
+    scope.updateAddress = ->
+      scope.edit_address_form.$setPristine()
+      if scope.edit_address_form.$valid
+        Customers.update(scope.address, scope.customer, scope.current_address).$promise.then (data) ->
+          scope.customer = data
+          template.dialog('close')
+          StatusMessage.display('success', "Address updated successfully.")
```

Comment:
We should put all language strings visible to a user into `config/locales/en.yml`.